pkg/api/defaults.go

```diff
@@ -508,6 +512,83 @@ func (p *Properties) setHostedMasterProfileDefaults() {
 	p.HostedMasterProfile.Subnet = DefaultKubernetesMasterSubnet
 }
 
+func (p *Properties) setCustomCloudProfileDefaults() {
+
+	if p.IsAzureStackCloud() {
+		//Set default value for ResourceManagerVMDNSSuffix
+		AzureStackCloudSpec.EndpointConfig.ResourceManagerVMDNSSuffix = p.CustomCloudProfile.Environment.ResourceManagerVMDNSSuffix
+		AzureCloudSpecEnvMap[AzureStackCloud] = AzureStackCloudSpec
+
+		// Use the custom input to overwrite the default values in AzureStackCloudSpec
+		if p.CustomCloudProfile.AzureEnvironmentSpecConfig != nil {
+			if len(p.CustomCloudProfile.AzureEnvironmentSpecConfig.CloudName) != 0 {
```

Comment:
@tariq1890 Any thoughts on len != 0 vs string != "" ? Not having researched, I would imagine you're using the  length non-equality check because it's more efficient @honcao ? If so, why not len(string) > 0. Wouldn't the gt/lt check be more efficient compared to an non-equality check?

Reply:
I think either are fine. We can go with what we use usually do in aks-engine for consistency's sake. I am more of a `string !=""` guy personally.

Reply:
Great, let's prefer `!= ""` for readability and more clear communication of intent